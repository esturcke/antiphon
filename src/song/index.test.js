// @flow
import fs from "mz/fs"
import { fromString } from "./index"

const path = (name: string) => `./test-data/songs/${name}`
;["empty", "single-line", "simple-song", "with-capo"]
  .map(name => path(name))
  .forEach(file =>
    it(`parses ${file}`, () =>
      fs
        .readFile(file, "utf-8")
        .then(raw => expect(fromString(raw)).toMatchSnapshot()))
  )
