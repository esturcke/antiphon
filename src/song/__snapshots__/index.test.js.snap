exports[`test parses ./test-data/songs/empty 1`] = `
Object {
  "artists": Array [],
  "capo": 0,
  "raw": "",
  "stanzas": Array [],
  "title": "",
}
`;

exports[`test parses ./test-data/songs/simple-song 1`] = `
Object {
  "artists": Array [
    "And an artist",
  ],
  "capo": 1,
  "raw": "This is the title
And an artist

Capo 1

Intro
C           Am           F           G
This is not a real song, but we use it anyway
C           Am           F           G
This is not a real song, but we use it anyway

Verse 1
C           Am           F           G
This is not a real song, but we use it anyway
C           Am           F           G
This is not a real song, but we use it anyway

Chorus
C           Am           F           G
This is not a real song, but we use it anyway
C           Am           F           G
This is not a real song, but we use it anyway

Verse 2
C           Am           F           G
This is not a real song, but we use it anyway
C           Am           F           G
This is not a real song, but we use it anyway

Chorus

Bridge
C           Am           F           G
This is not a real song, but we use it anyway
C           Am           F           G
This is not a real song, but we use it anyway",
  "stanzas": Array [
    Object {
      "id": undefined,
      "lines": Array [
        Array [],
        Array [],
        Array [],
      ],
      "type": "verse",
    },
    Object {
      "id": 1,
      "lines": Array [
        Array [],
        Array [],
      ],
      "type": "verse",
    },
    Object {
      "id": undefined,
      "lines": Array [
        Array [],
        Array [],
      ],
      "type": "chorus",
    },
    Object {
      "id": 2,
      "lines": Array [
        Array [],
        Array [],
      ],
      "type": "verse",
    },
    Object {
      "id": undefined,
      "lines": Array [],
      "type": "chorus",
    },
    Object {
      "id": undefined,
      "lines": Array [
        Array [],
        Array [],
      ],
      "type": "bridge",
    },
  ],
  "title": "This is the title",
}
`;

exports[`test parses ./test-data/songs/single-line 1`] = `
Object {
  "artists": Array [],
  "capo": 0,
  "raw": "This is the title",
  "stanzas": Array [],
  "title": "This is the title",
}
`;

exports[`test parses ./test-data/songs/with-capo 1`] = `
Object {
  "artists": Array [
    "The artist",
  ],
  "capo": 3,
  "raw": "The title
The artist

Capo 3",
  "stanzas": Array [],
  "title": "The title",
}
`;
